packages/trpc/server/routers/viewer/workflows.tsx

```diff
@@ -49,32 +51,149 @@ function getSender(
   return isSMSAction(step.action) ? step.sender || SENDER_ID : step.senderName || SENDER_NAME;
 }
 
+async function isAuthorized(
+  workflow: Pick<Workflow, "id" | "teamId" | "userId"> | null,
+  prisma: PrismaClient<
+    Prisma.PrismaClientOptions,
+    never,
+    Prisma.RejectOnNotFound | Prisma.RejectPerOperation | undefined
+  >,
+  currentUserId: number
+) {
+  if (!workflow) {
+    return false;
+  }
+
+  const userWorkflow = await prisma.workflow.findFirst({
+    where: {
+      id: workflow.id,
+      OR: [
+        { userId: currentUserId },
+        {
+          team: {
+            members: {
+              some: {
+                userId: currentUserId,
+              },
+            },
+          },
+        },
+      ],
+    },
+  });
+
+  if (userWorkflow) return true;
+
+  return false;
+}
+
 export const workflowsRouter = router({
-  list: authedProcedure.query(async ({ ctx }) => {
-    const workflows = await ctx.prisma.workflow.findMany({
-      where: {
-        userId: ctx.user.id,
-      },
-      include: {
-        activeOn: {
-          select: {
-            eventType: {
+  list: authedProcedure
+    .input(
+      z
+        .object({
+          teamId: z.number().optional(),
+          userId: z.number().optional(),
+        })
+        .optional()
+    )
+    .query(async ({ ctx, input }) => {
```

Comment:
It returns the team workflow for a team member that hasn't accepted the invitation.